io-engine/src/rebuild/rebuild_task.rs

```diff
@@ -110,58 +123,21 @@ impl RebuildTask {
     /// Copies one segment worth of data from source into destination.
     async fn copy_one(
         &mut self,
-        blk: u64,
-        descriptor: &RebuildDescriptor,
+        offset_blk: u64,
+        desc: &RebuildDescriptor,
     ) -> Result<(), RebuildError> {
-        let mut copy_buffer: DmaBuf;
-        let source_hdl = descriptor.src_io_handle().await?;
-        let destination_hdl = descriptor.dst_io_handle().await?;
+        let buf_lock = self.buffer.lock().await;
```

Comment:
Why does the buffer need a lock? we have a mutable ref here so we can modify it with extra indirection?

Reply:
Ah, indeed we don't need a lock here. I didn't notice that each task is already locked when called from RebuildTasks. Will fix.